nsot/models/network.py

```diff
@@ -588,3 +615,17 @@ def to_dict(self):
             'state': self.state,
             'attributes': self.get_attributes(),
         }
+
+
+# Signals
+def refresh_assignment_interface_networks(sender, instance, **kwargs):
```

Comment:
A brief docstring explaining the purpose of this method would be good.

Also should see if a fetch of related fields for each assignment in a single query might be better. I'm concerned that enumerating all child networks for a parent network and then each assignment for each interface may result in a lot of small queries to the database.

I'm thinking something like:

```
    for child in instance.children.all():
        for assignment in child.assignments.select_related('interface'):
            assignment.interface.clean_addresses()
            assignment.interface.save()
```

In practice each network is typically only ever assigned to one interface so this would reduce each assignment lookup to a single query vs. two (1 for assignment, 1 for the interface).

I'm just thinking ahead. If it feels like this is pre-optimizing, at the least, it would be wise to make a note in the comments that this could be a potential bottleneck for networks with many children that are assigned to interfaces. If such a bottleneck does popup in production, you know where to look.

Reply:
Additionally, my suggestion to leverage `.select_related()` might not be comprehensive enough. Especially because `Interface.clean_addresses()` is actually doing at least two more queries internally prior to the explicit call to `Interface.save()` which results in one more query.

Reply:
I'll definitely add in the docstring. So given that `.select_related()` isn't that comprehensive, do you think I should leave it as is for now or implement it with `.select_related()` ?